Allow user defined 'Accept' and 'Content-Type' headers

Peak-CDE opened the pull request:
This allows users to override the 'Accept' and 'Content-Type' headers that were previously hard coded, as if you added them yourself they would just be appended. Some applications such as [PayPal's access token](https://developer.paypal.com/docs/api/get-an-access-token-curl/) request were impossible to use as the hard coded header took prescience over the one added by the plugin even though they were both sent. The default values are still set, but now they are set in the same way as the other headers so they can be overridden.

ErikMinekus commented:
Thanks! This looks fine for JSON requests, but `DownloadFile` and `UploadFile` will send two `Accept` and `Content-Type` headers now. Perhaps we can simply call `SetHeader` in those methods, since I don't think there's a need to set custom `Accept` and `Content-Type` headers for file transfers.

Peak-CDE commented:
> Thanks! This looks fine for JSON requests, but `DownloadFile` and `UploadFile` will send two `Accept` and `Content-Type` headers now. Perhaps we can simply call `SetHeader` in those methods, since I don't think there's a need to set custom `Accept` and `Content-Type` headers for file transfers.

Good point, totally overlooked those. I replaced them with SetHeader in Download/Upload methods as well and it seems to work properly. I also moved SetHeader out of the constructor and back into the Perform method to make it more consistent with the new upload/download methods, and that means we can go back to the old constructor as well.

Peak-CDE commented:
And I just realized that commit would break the thing I was trying to fix in the first place, so back to using the new constructor it is.
